### PR TITLE
Set the opam CLI version as 2.1 manually when build Pyre(and Pysa)

### PR DESCRIPTION
**Pull request by WangGithubUser**
**Pre-submission checklist**
- [x] I've ran the linters locally and fixed lint errors related to the files I modified in this PR. You can install the linters by running `pip install -r requirements-dev.txt && pre-commit install`
- [x] `pre-commit run`

## Summary

Detail at GH-752
## Test Plan
Before this PR:
image
After this PR:
image

### CODE REVIEW

**Comment by arthaud** [comment]
Thanks for doing this.

I see this warning in the tests: https://github.com/facebook/pyre-check/actions/runs/5540659955/jobs/10114535419?pr=753#step:7:27
Maybe it's better to not set `OPAMCLI` but provide `--cli=2.1` to all commands?

**Comment by WangGithubUser** [comment]
@arthaud I'm sorry that missed this.

File: .github/workflows/pysa.yml [hunk 1]
```diff
@@ -33,6 +33,7 @@ jobs:
           echo "CAML_LD_LIBRARY_PATH=$HOME/.opam/4.14.0/lib/stublibs:$HOME/.opam/4.14.0/lib/ocaml/stublibs:$HOME/.opam/4.14.0/lib/ocam" >> $GITHUB_ENV
           echo "$HOME/.opam/4.14.0/bin" >> $GITHUB_PATH
           echo "/home/opam/.opam/4.14.0/bin" >> $GITHUB_PATH
+          echo "OPAMCLI=2.1" >> $GITHUB_ENV
```
Comment (arthaud) [inline .github/workflows/pysa.yml:36]: Could you remove that line? It shouldn't be required

Comment (WangGithubUser) [reply .github/workflows/pysa.yml:36]: I already tried before, but it failed.

Comment (WangGithubUser) [reply .github/workflows/pysa.yml:36]: Perhaps the priority of environment variables is higher than the option `-- cli=2.1`?

Comment (arthaud) [reply .github/workflows/pysa.yml:36]: As I said above, this get the following warning:
```
Warning:  OPAMCLI should only ever be set to 2.0 - use '--cli=2.1' instead.
```
It sounds like `OPAMCLI=2.1` might not work correctly.

**Review by arthaud — APPROVED** [review]

**Review by arthaud — APPROVED** [review]

**Comment by arthaud** [comment]
Unfortunately it fails in our internal CI, I will need more time to look at it..

**Comment by WangGithubUser** [comment]
The problem has resolved in b9386eb42d19eaf51c052d9c52d5044d9d10b235, I will close this PR.

**Comment by WangGithubUser** [comment]
@arthaud The check is still failing, can you have a look for #755 and #754 ?

**Comment by arthaud** [comment]
Yes I will look at it.

**Pull request closed without merging**